### ISSUE TITLE
Undo ignores last changes when cursor changes context

### ISSUE THREAD

**Issue by kwilcz** [getinsomnia/insomnia#7671]
### Expected Behavior

After making a change, Command + Z (CTRL + Z) should undo any last update.

### Actual Behavior

When a change is made, if cursor is moved to any other textbox, undo action does nothing.

### Reproduction Steps

1. Create new HTTP request
2. Enter any text in request URL textbox
3. Click (focus cursor) on new parameter name textbox. No need to type anything
4. Click CTRL + Z and notice that nothing is happening

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

Undo action is expected to undo any change recently made in the editor / app. Current behavior may cause data to be lost, as more often then not some quick changes are done to requests which cannot be easily undone.

### Insomnia Version

9.3.2

### What operating system are you using?

Windows

### Operating System Version

Windows 11 version 23h2, build 22631.3593

### Installation method

insomnia.rest installer

**Comment by ihexxa** [comment]
Hello @kwilcz , I would like to double check, in your test case, do you expect to undo changes of the URL textbox or changes of the parameter name textbox?

**Comment by ihexxa** [comment]
Nvm it seems to be the parameter name textbox.

**Comment by kwilcz** [comment]
@ihexxa Also it can be noticed that every text box has it's own undo history. It is fine unless you want to revert all changes up to a specific point. Then instead of just clicking `CTRL + Z` multiple times, you'd have to click through all inputs and undo on every single one of them.

After some additional tests, I think this is the main issue I detected as when I click back to the URL, I can revert changes made there.

**Comment by ihexxa** [comment]
> Also it can be noticed that every text box has it's own undo history

This is true, I also agree that introducing sort of "global" history would be nice, which could be paired with an operation history pane, while we still need to evaluate it's effort and ux changes, thanks @kwilcz for sharing thoughts.
